pkg/auth/oauth/external/handler.go

```diff
@@ -214,8 +214,7 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 }
 
 func (h *Handler) handleError(err error, w http.ResponseWriter, req *http.Request) {
-	handled, err := h.errorHandler.AuthenticationError(err, w, req)
-	if handled {
+	if handled, _ := h.errorHandler.AuthenticationError(err, w, req); handled {
```

Comment:
this implementation will cause different behavior, it will no longer write the header